src/pybind/mgr/dashboard/services/sso.py

```diff
@@ -79,20 +66,30 @@ def load(cls):
         json_db = mgr.get_store(cls.ssodb_config_key(), None)
         if json_db is None:
             logger.debug("No DB v%s found, creating new...", cls.VERSION)
-            db = cls(cls.VERSION, '', Saml2({}))
+            db = cls(cls.VERSION, AuthType.LOCAL, Saml2({}))
             # check if we can update from a previous version database
             db.check_and_update_db()
             return db
 
-        dict_db = json.loads(json_db)  # type: dict
-        return cls(dict_db['version'], dict_db.get('protocol'),
-                   Saml2.from_dict(dict_db.get('saml2')))
+        dict_db = json.loads(json_db)  # type: Dict[str, str]
+        protocol: AuthType = AuthType(str(dict_db.get('protocol')))
+        config: SSOAuth = (Saml2.from_dict(str(dict_db.get('saml2'))) if protocol == AuthType.SAML2
+                           else OAuth2.from_dict(str(dict_db.get('oauth2'))))
+        return cls(dict_db['version'], protocol, config)
 
 
 def load_sso_db():
     mgr.SSO_DB = SsoDB.load()  # type: ignore
 
 
+@CLIWriteCommand("dashboard sso enable oauth2")
+def enable_sso(_):
+    mgr.SSO_DB.protocol = AuthType.OAUTH2
+    mgr.SSO_DB.save()
+    mgr.set_module_option('sso_oauth2', True)
+    return HandleCommandResult(stdout='SSO is "enabled" with "OAuth2" protocol.')
```

Comment:
No disable command?

What I did, for example, for the debug command is making the enable/disable the parameter of the CLI command (this way you only need 1 command instead of 3, but I'm not sure if you tried to make this compatible with some `ceph dashboard sso enable saml2` command):
```sh
ceph dashboard debug enable
ceph dashboard debug disable
ceph dashboard debug status
```

Reply:
There is no disable because it is already implemented as[ `dashboard sso disable` in the saml2 sso cmds], so I just modied that a bit(https://github.com/ceph/ceph/pull/58456/files#diff-51ae22ae2c7aded6a4a8d414641c20b835bba5a810630b55c8ad9dc9eda9451cL150)

Reply:
Ah ok, got it!